kerastuner/tuners/ultraband.py

```diff
@@ -12,18 +12,210 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"Variation of HyperBand algorithm."
 
-from ...engine import tuner as tuner_module
-from ...engine import oracle as oracle_module
+import copy
+import queue
+import random
+from ..engine import tuner as tuner_module
+from ..engine import execution as execution_module
+from ..engine import tuner_utils
+from ..engine import oracle as oracle_module
 
 
 class UltraBandOracle(oracle_module.Oracle):
-    # TODO
-    pass
+    """ Oracle class for UltraBand.
+
+    # Attributes:
+        trails: An integer. The maximum number of trails allowed.
+        queue: An instance of Queue. The elements in the queue are neural network candidate indices.
+
+    """
+
+    def __init__(self,
+                 trials=200,
+                 seed=None,
+                 factor=3,
```

Comment:
As a general best practice, constructor args should be accessible on the class instance (`x.factor`)